packages/next/src/lib/metadata/get-metadata-route.ts

```diff
@@ -11,14 +11,33 @@ import { normalizePathSep } from '../../shared/lib/page-path/normalize-path-sep'
  * Give it a unique hash suffix to avoid conflicts
  *
  * e.g.
- * /app/opengraph-image.tsx -> /opengraph-image
- * /app/(post)/opengraph-image.tsx -> /opengraph-image-[0-9a-z]{6}
+ * /opengraph-image -> /opengraph-image
+ * /(post)/opengraph-image.tsx -> /opengraph-image-[0-9a-z]{6}
+ *
+ * Sitemap is an exception, it should not have a suffix.
+ * As the generated urls are for indexer and usually one sitemap contains all the urls of the sub routes.
+ * The sitemap should be unique in each level and not have a suffix.
+ *
+ * /sitemap -> /sitemap
+ * /(post)/sitemap -> /sitemap
  */
 function getMetadataRouteSuffix(page: string) {
-  let suffix = ''
+  // Remove the last segment and get the parent pathname
+  // e.g. /parent/a/b/c -> /parent
+  // e.g. /parent/opengraph-image.tsx -> /parent
+  const parentPathname = page.slice(0, -(path.basename(page).length + 1))
+  // Only apply suffix to metadata routes except for sitemaps
+  if (page.endsWith('/sitemap')) {
+    return ''
+  }
 
-  if ((page.includes('(') && page.includes(')')) || page.includes('@')) {
-    suffix = djb2Hash(page).toString(36).slice(0, 6)
+  // Calculate the hash suffix based on the parent path
+  let suffix = ''
+  if (
+    (parentPathname.includes('(') && parentPathname.includes(')')) ||
+    parentPathname.includes('@')
```

Comment:
This is looking for parallel routes right? I think this should probably be `startsWith`, since it's technically possible to have a route with `@` that doesn't correspond with a parallel route. 

```suggestion
    parentPathname.startsWith('@')
```

Reply:
This is similar as the group segment one, we check if any special char exists in the pathname

Reply:
Is that correct though? It’s only considered special if it’s at the start of the dirname. It’s valid to have a directory called “foo@“ that has nothing to do with parallel routes.